src/Decorators/model.ts

```diff
@@ -0,0 +1,21 @@
+/* eslint-disable @typescript-eslint/ban-types */
+import { setModelName, addOptions } from './shared/model-service';
+import { ModelClassDecoratorOptions } from './shared/data-types';
+
+export function Model(options?: ModelClassDecoratorOptions): Function;
+export function Model(target: Function): void;
+export function Model(arg: any): void | Function {
```

Comment:
I'll see what I can do

Reply:
So I messed around with it for a minute and I think my hands are tied in that instance. I went with that way of implementing it so that I could support the two different overrides. That said, I can constrain it with an unknown type to force me to validate it later on.

Reply:
You can check out the new impl I went with